x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/controller.go

```diff
@@ -65,38 +68,51 @@ func NewController(
 }
 
 // Update applies config change and performes all steps necessary to apply it.
-func (e *Controller) Update(c *config.Config) error {
-	if err := info.InjectAgentConfig(c); err != nil {
-		return err
+func (e *Controller) Update(ctx context.Context, c *config.Config) (rErr error) {
+	span, ctx := apm.StartSpan(ctx, "update", "app.internal")
+	defer func() {
+		if rErr != nil {
+			apm.CaptureError(ctx, rErr).Send()
+		}
```

Comment:
[Suggestion]
It's safe to call `apm.CaptureError` with a nil error and then call `Send`. Therefore it can be simplified.
```suggestion
		apm.CaptureError(ctx, rErr).Send()
```